flang/lib/Optimizer/Transforms/OMPMapInfoFinalization.cpp

```diff
@@ -219,6 +393,31 @@ class OMPMapInfoFinalizationPass
     mapClauseOwner.getMapVarsMutable().assign(newMapOps);
   }
 
+  // We retrieve the first user that is a Target operation, there
+  // should only be one currently, every MapInfoOp can be tied to
+  // at most 1 Target operation and at the minimum no operation,
+  // this may change in the future with IR cleanups/modifications
+  // in which case this pass will need updated to support cases
+  // where a map can have more than one user and more than one of
+  // those users can be a Target operation. For now, we simply
+  // return the first target operation encountered, which may
+  // be on the parent MapInfoOp in the case of a member mapping
+  // in which case we must traverse the MapInfoOp chain until we
+  // find the first TargetOp user.
```

Comment:
Nit: Also re-flow comments to extend to 80 characters.
```suggestion
  // We retrieve the first user that is a Target operation, of which there
  // should only be one currently. Every MapInfoOp can be tied to
  // at most one Target operation and at the minimum no operations.
  // This may change in the future with IR cleanups/modifications,
  // in which case this pass will need updating to support cases
  // where a map can have more than one user and more than one of
  // those users can be a Target operation. For now, we simply
  // return the first target operation encountered, which may
  // be on the parent MapInfoOp in the case of a member mapping.
  // In that case, we traverse the MapInfoOp chain until we
  // find the first TargetOp user.
```